docs/snowpark/snowpark-backend-joins.md

```diff
@@ -0,0 +1,178 @@
+simpleInnerJoin dataSetA dataSetB =
+     dataSetA
+        |> innerJoin dataSetB
+            ( \a b -> a.id == b.id )
+        |> List.map (\ (x, y) -> 
+            { idA = x.id
+            , idB = y.id
+            })
+```
+
+The `simpleInnerJoin` example illustrates how to perform an *inner join* operation between two datasets, dataSetA and dataSetB, represented as lists of types TypeA and TypeB respectively. In this case, the join is based on a specific condition: equality between the `id` properties of elements in dataSetA and dataSetB.
+
+The `innerJoin` function (provided for the *Morphir.SDK*) facilitates this join, taking the equality condition provided in the lambda function `(\a b -> a.id == b.id)`.
+
+Subsequently, the List.map function is used to project the results of the join. The lambda function `(\(x, y) -> { idA = x.id, idB = y.id })` maps each pair of results (x, y) to a new data type `{ idA : Int, idB : Int }`, where `idA` represents the `id` value in dataSetA and `idB` represents the `id` value in dataSetB.
+
+This example demonstrates how Morphir in Elm can be used concisely and expressively to perform join and projection operations, providing a solid foundation for efficient data manipulation.
+
+## Joins in Snowpark
+
+In Snowpark, support is provided for specific scenarios that follow the join pattern.
+
+### Inner Join
+
+If you have this code in ELM using the `innerJoin` function:
+
+```elm
+simpleInnerJoin : List TypeA -> List TypeB -> List {idA: Int, idB: Int}
+simpleInnerJoin dataSetA dataSetB =
+     dataSetA
+        |> innerJoin dataSetB
+            ( \a b -> a.id == b.id )
+        |> List.map (\ (x, y) -> 
+            { idA = x.id
+            , idB = y.id
+            })
+```
+
+The transformation to Snowpark will be:
+
+```Scala
+    def simpleInnerJoin
+        (dataSetA: com.snowflake.snowpark.DataFrame)
+        (dataSetB: com.snowflake.snowpark.DataFrame)
+        (implicit sfSession: com.snowflake.snowpark.Session): com.snowflake.snowpark.DataFrame = {
+    val dataSetAColumns: deparments.Joins.TypeA = new deparments.Joins.TypeAWrapper(dataSetA)    
+    val dataSetBColumns: deparments.Joins.TypeB = new deparments.Joins.TypeBWrapper(dataSetB)
+    
+    dataSetA.join(
+      dataSetB,
+      (dataSetAColumns.id) === (dataSetBColumns.id),
+      "inner"
+    ).select(
+      dataSetAColumns.id.alias("idA"),
+      dataSetBColumns.id.alias("idB")
+    )
+  }
+```
+
+You can notice that `innerJoin` function from **Morphir.SDK** is transform to `join` function of **Snowpark** passing the argument `inner` to choose the join type.
+
+The projection is change to a **Snowpark select** to return a dataframe with the selected columns.
+
+### Left Join
+
+If you have this code in ELM using the `leftJoin` function is similar to the `innerJoin` function. The different is that the `leftJoin` returns a `List ( a, Maybe b)` instead, so in the projection you must to return a maybe value:
```

Comment:
The first phrase "If you have this code in ELM... to the `innerJoin` function." its a little bit confusing.

 Maybe we can change it to say that "The `leftJoin` function is similar to `innerJoin`" (we can provide links to the documentation of these functions: https://package.elm-lang.org/packages/finos/morphir-elm/latest/Morphir-SDK-List#leftJoin ) . 

Also change "The different is that..." to "The difference is that..."